core/src/main/java/de/danoeh/antennapod/core/feed/SubscriptionsFilter.java

```diff
@@ -0,0 +1,116 @@
+package de.danoeh.antennapod.core.feed;
+
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+import de.danoeh.antennapod.core.util.LongIntMap;
+
+public class SubscriptionsFilter {
+    private final String[] properties;
+
+    private boolean showIfCounterGreaterZero = false;
+
+    private boolean showAutoDownloadEnabled = false;
+    private boolean showAutoDownloadDisabled = false;
+
+    private boolean showUpdatedEnabled = false;
+    private boolean showUpdatedDisabled = false;
+
+    public SubscriptionsFilter(String properties) {
+        this(TextUtils.split(properties, ","));
+    }
+
+
+    public SubscriptionsFilter(String[] properties) {
+        this.properties = properties;
+        for (String property : properties) {
+            // see R.arrays.feed_filter_values
+            switch (property) {
+                case "counter_greater_zero":
+                    showIfCounterGreaterZero = true;
+                    break;
+                case "enabled_auto_download":
+                    showAutoDownloadEnabled = true;
+                    break;
+                case "disabled_auto_download":
+                    showAutoDownloadDisabled = true;
+                    break;
+                case "enabled_updates":
+                    showUpdatedEnabled = true;
+                    break;
+                case "disabled_updates":
+                    showUpdatedDisabled = true;
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    public boolean areSubscriptionsFiltered() {
+        return properties.length > 0;
+    }
+
+    /**
+     * Run a list of feed items through the filter.
+     */
+    public List<Feed> filter(List<Feed> items, PodDBAdapter adapter) {
```

Comment:
Please pass the `feedCounters` array here instead of the database adapter. This avoids having to load the counters twice. Every caller of the method queries the counters, anyway.